tools/importer-rest-api-specs/components/schema/fields_resource_id.go

```diff
@@ -91,12 +90,17 @@ func (b Builder) identifyTopLevelFieldsWithinResourceID(input resourcemanager.Re
 
 			userConfigurableSegments++
 			fieldName := strings.Title(v.Name)
-			hclName := helpers.ConvertToSnakeCase(v.Name)
 			if i == len(input.Segments)-1 {
 				// if it's the last one override the name since that'll be the name of this Resource
 				fieldName = "Name"
-				hclName = "name"
 			}
+			var err error
+			fieldName, err = updateFieldNameFromSchemaOverrides(fieldName, resource)
```

Comment:
The current implementation supports overrides for Resource IDs and nested fields, it doesn't for top level fields because [this method](https://github.com/hashicorp/pandora/blob/bcd0d69d614a268ced02a2f0277800fbd3d0a5bd/tools/importer-rest-api-specs/components/schema/helpers.go#L61-L70) that you linked isn't called for top level fields and by the looks of it shouldn't be for the moment since it's mainly hardcoding the field details and mappings for properties like location, zones, identity etc.

I disagree about reusing that method on the fields we source from the resource IDs, my reason is explained in point three of the PR description. As a compromise I've removed the schema overrides as a field processor, rewritten this function to `applySchemaOverrides` to do the overrides. `applySchemaOverrides` is called both here and by the method linked in the paragraph above, so that nested fields can have overrides applied too.